### DISCUSSION TITLE
How does datafuse manage metadata or catalog?

### DISCUSSION THREAD

**Discussion by leiysky** [databendlabs/databend#3358] [Q&A] [answered]

**Comment by bohutang** [comment]
FuseQuery is stateless, the metadata and catalog all is from FuseStore by flight rpc, but has a local cache in a general sense.

**Comment by drmingdrmer** [comment, marked as answer]
There is a global consensus group(something like a raft cluster) storing all metadata such as databases, table schemas, etc. A store server subscribes meta changes by participating in the raft cluster as a **learner**. There are up to 5 servers that act as raft **candidate** participants in a store cluster. 

The consistency level between candidates and learners is eventual consistency, e.g., a write may not be seen by the following read if the read RPC is sent to a follower.
